### PR TITLE
[oneMKL-BLAS] Update precision support for gemm, gemm batch, rot

### PR DESCRIPTION
**Pull request by b-shi**
This PR updates the precision support tables for `gemm`, `gemm_batch`, and `rot` BLAS routines with newly supported precisions.

### CODE REVIEW

**Comment by b-shi** [comment]
Hi @mmeterel, could you please take a quick look at these changes?

File: source/elements/oneMKL/source/domains/blas/rot.rst [hunk 1]
```diff
@@ -43,6 +55,10 @@ the sum of two of these scalar-vector products as follow:
         -  ``float`` 
       * -  ``double`` 
         -  ``double`` 
+      * -  ``std::complex<float>``
```
Comment (mmeterel) [inline source/elements/oneMKL/source/domains/blas/rot.rst:58]: I see difference with DPC++ documentation, where we support seperate T_scalarC and T_scalarS.

File: source/elements/oneMKL/source/domains/blas/gemm.rst [hunk 2]
```diff
@@ -45,6 +45,14 @@ op(``X``) = ``X``\ :sup:`H`,
        -  Ta 
        -  Tb 
        -  Tc 
+     * -  ``float``
```
Comment (mmeterel) [inline source/elements/oneMKL/source/domains/blas/gemm.rst:48]: Can we match to DPC++ documentation to match the table titles for consistency? (Ta (A Matrices))....

File: source/elements/oneMKL/source/domains/blas/rot.rst (hunk 1 above)
Comment (b-shi) [reply source/elements/oneMKL/source/domains/blas/rot.rst]: Oops, thanks for the catch, fixed.

File: source/elements/oneMKL/source/domains/blas/gemm.rst (hunk 2 above)
Comment (b-shi) [reply source/elements/oneMKL/source/domains/blas/gemm.rst]: Sure, I've reordered the columns to match the DPC++ docs.

Comment (mmeterel) [reply source/elements/oneMKL/source/domains/blas/gemm.rst]: i meant the table titles are different.

Comment (b-shi) [reply source/elements/oneMKL/source/domains/blas/gemm.rst]: Ah okay, updated ^

**Review by mmeterel — APPROVED** [review]
Looks good. Thanks for the PR!

**Pull request merged**